### PR TITLE
fakeroute: 0.2 -> 0.3

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Description of changes

Fakeroute has been unmantained for a very long time and recently the source
code was close to being lost. Morover, the original program has not been working
for a while due to a combination of C++ breaking changes and pre-existing bugs.
So, I adopted the project, re-created the [git repository](https://maxwell.ydns.eu/git/rnhmjoj/fakeroute), fixed all the issues I could
find and made a new release.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested via `fakeroute.tests`
- [x] Tested compilation of all packages that depend on this change
- [x] Tested basic functionality of all binary files
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/tests/fakeroute.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import ./make-test-python.nix ({ pkgs, ...} : {
```
Comment (AndersonTorres) [inline nixos/tests/fakeroute.nix:1]:
```suggestion
import ./make-test-python.nix ({ lib, pkgs, ...} : {
```

File: nixos/tests/fakeroute.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  name = "fakeroute";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ rnhmjoj ];
+  };
```
Comment (AndersonTorres) [inline nixos/tests/fakeroute.nix:5]:
```suggestion
  meta.maintainers = with lib.maintainers; [ rnhmjoj ];
```

File: pkgs/tools/networking/fakeroute/default.nix [hunk 3]
```diff
@@ -1,25 +1,22 @@
...
   pname = "fakeroute";
   version = "0.3";
 
-  src = fetchFromGitHub {
-    owner = "museoa";
-    repo = "fakeroute";
-    rev = "f8cb9c0ae3abb4c0662d9e1fcac67eefeeac3963";
-    sha256 = "12dhahwlpjzv79wpdpryjihamfbh4d8cfzfw4wi1jkl0dv2d41jg";
+  src = fetchurl {
+    url = "https://maxwell.ydns.eu/git/rnhmjoj/${pname}/releases/download/v${version}/${pname}-${version}.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/tools/networking/fakeroute/default.nix:8]: Do not parameterize pname.

File: pkgs/tools/networking/fakeroute/default.nix [hunk 4]
```diff
@@ -1,25 +1,22 @@
...
+    sha256 = "DoXGJm8vOlAD6ZuvVAt6bkgfahc8WgyYIXCrgqzfiWg=";
```
Comment (AndersonTorres) [inline pkgs/tools/networking/fakeroute/default.nix:9]:
```suggestion
    hash = "sha256-DoXGJm8vOlAD6ZuvVAt6bkgfahc8WgyYIXCrgqzfiWg=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Why the compilation error on Darwin Intel?

**Comment by rnhmjoj** [comment]
Apparently half of the socket APIs in BSD, linux and darwin are slightly incompatible.
For now it will stay linux only.

**Comment by phanirithvij** [comment]
the git repo is dead?

**Comment by rnhmjoj** [comment]
That was fast: I lost control of my domain a few days ago and had to set up a new one. I'll update the url.

**Comment by phanirithvij** [comment]
I guess p2p/magnetico too.

I found this via coincidence trough [here](https://discourse.nixos.org/t/the-nixos-foundations-call-to-action-s3-costs-require-community-support/28672/27)

> control of my domain a few days ago

I remember seeing your message on matrix moderation channel.

**Comment by rnhmjoj** [comment]
Yes, I just updated both. See https://github.com/NixOS/nixpkgs/pull/350662.

**Pull request merged**